lib/quickpay/api/client.rb

```diff
@@ -42,29 +50,48 @@ def initialize(username: nil, password: nil, base_uri: "https://api.quickpay.net
           end
 
           req = Request.new(
-            method,
+            method.to_sym,
             path,
             scrub_body(body.dup, headers["Content-Type"]),
             headers,
-            options.fetch(:query, {})
+            options[:query]
           ).freeze
 
-          res = @connection.request(**req.to_h)
-          error = QuickPay::API::Error.by_status_code(res.status, res.body, res.headers, req)
+          uri = URI(@base_uri)
+          uri.path << req.path
+          if (query = req.query) && query.any?
+            uri.query = URI.encode_www_form(req.query)
+          end
+          net_req = method_class.new(uri, req.headers)
+          net_req.basic_auth(@username, @password) if @username || @password
+          net_req.body = req.body
+          res = Net::HTTP.start(
+            uri.hostname,
+            use_ssl: uri.scheme == HTTPS,
+            open_timeout: @connect_timeout,
+            read_timeout: @read_timeout,
+            write_timeout: @write_timeout
+          ) do |http|
+            http.request(net_req)
+          end
+          status_code = res.code.to_i
+          body = res.body
+          headers = res.each_header.to_h
```

Comment:
There's a pretty big breaking change here, if we can't guarantee the headers "upper/lower-caseness".

Perhaps we could consider returning a Hash-like object that behaves like [`Net::HTTPHeader`](https://github.com/ruby/net-http/blob/master/lib/net/http/header.rb), because otherwise stuff might break when we or externals upgrade to version 4.

Reply:
Which is the reason for the major version change.

Reply:
Besides changes in underlying client errors

Reply:
shouldn't you be sleeping